config/v1alpha1/types_backup.go

```diff
@@ -0,0 +1,164 @@
+	Schedule string `json:"schedule"`
+
+	// Cron Regex breakdown:
+	// Allow macros: (@(annually|yearly|monthly|weekly|daily|hourly))
+	// OR
+	// Minute:
+	//   (\*|(?:\*|(?:[0-9]|(?:[1-5][0-9])))\/(?:[0-9]|(?:[1-5][0-9]))|(?:[0-9]|(?:[1-5][0-9]))(?:(?:\-[0-9]|\-(?:[1-5][0-9]))?|(?:\,(?:[0-9]|(?:[1-5][0-9])))*))
+	// Hour:
+	//   (\*|(?:\*|(?:\*|(?:[0-9]|1[0-9]|2[0-3])))\/(?:[0-9]|1[0-9]|2[0-3])|(?:[0-9]|1[0-9]|2[0-3])(?:(?:\-(?:[0-9]|1[0-9]|2[0-3]))?|(?:\,(?:[0-9]|1[0-9]|2[0-3]))*))
+	// Day of the Month:
+	//   (\*|(?:[1-9]|(?:[12][0-9])|3[01])(?:(?:\-(?:[1-9]|(?:[12][0-9])|3[01]))?|(?:\,(?:[1-9]|(?:[12][0-9])|3[01]))*))
+	// Month:
+	//   (\*|(?:[1-9]|1[012]|JAN|FEB|MAR|APR|MAY|JUN|JUL|AUG|SEP|OCT|NOV|DEC)(?:(?:\-(?:[1-9]|1[012]|JAN|FEB|MAR|APR|MAY|JUN|JUL|AUG|SEP|OCT|NOV|DEC))?|(?:\,(?:[1-9]|1[012]|JAN|FEB|MAR|APR|MAY|JUN|JUL|AUG|SEP|OCT|NOV|DEC))*))
+	// Day of Week:
+	//   (\*|(?:[0-6]|SUN|MON|TUE|WED|THU|FRI|SAT)(?:(?:\-(?:[0-6]|SUN|MON|TUE|WED|THU|FRI|SAT))?|(?:\,(?:[0-6]|SUN|MON|TUE|WED|THU|FRI|SAT))*))
+	//
+
+	// The time zone name for the given schedule, see https://en.wikipedia.org/wiki/List_of_tz_database_time_zones.
+	// If not specified, this will default to the time zone of the kube-controller-manager process.
+	// See https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/#time-zones
+	// +kubebuilder:validation:Optional
+	// +optional
+	// +kubebuilder:validation:Pattern:=`^([A-Za-z_]+([+-]*0)*|[A-Za-z_]+(\/[A-Za-z_]+){1,2})(\/GMT[+-]\d{1,2})?$`
+	TimeZone string `json:"timeZone,omitempty"`
+
+	// Timezone regex breakdown:
+	// ([A-Za-z_]+([+-]*0)*|[A-Za-z_]+(/[A-Za-z_]+){1,2}) - Matches either:
+	//   [A-Za-z_]+([+-]*0)* - One or more alphabetical characters (uppercase or lowercase) or underscores, followed by a +0 or -0 to account for GMT+0 or GMT-0 (for the first part of the timezone identifier).
+	//   [A-Za-z_]+(/[A-Za-z_]+){1,2} - One or more alphabetical characters (uppercase or lowercase) or underscores, followed by one or two occurrences of a forward slash followed by one or more alphabetical characters or underscores. This allows for matching timezone identifiers with 2 or 3 parts, e.g America/Argentina/Buenos_Aires
+	// (/GMT[+-]\d{1,2})? - Makes the GMT offset suffix optional. It matches "/GMT" followed by either a plus ("+") or minus ("-") sign and one or two digits (the GMT offset)
+
+	// RetentionPolicy defines the retention policy for retaining and deleting existing backups.
+	// +kubebuilder:validation:Optional
+	// +optional
+	RetentionPolicy RetentionPolicy `json:"retentionPolicy"`
+
+	// PVCName specifies the name of the PersistentVolumeClaim which binds a PersistentVolume where the
+	// etcd backup files would be saved
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +required
+	PVCName string `json:"pvcName"`
```

Comment:
Allow "" and have it mean a rotating local volume, with one local volume per master.  This will allow a "do nothing" cluster to have automated backups with each backup on a different master to distribute risk.

Reply:
I think most people running backups already have a SAN/NAS in place which they would mount via NFS for such purposes. Maybe the "" default would trigger it the way it would be when you run the backup pod today, just on plain host mounts. 

Besides making this somewhat difficult to implement retention, how would you implement the rotation? Have a different node selector every time the backup triggers?

Reply:
Good question. I had that issue pointed out in the initial proposal which is why we started with the simpler PVC approach.
Since we're targeting the no-opinion local backup option as a follow up, I will do a follow up on the enhancement to try and sketch out how we could do local backups and the associated scheduling and retention challenges.

Reply:
> Besides making this somewhat difficult to implement retention, how would you implement the rotation? Have a different node selector every time the backup triggers?

You have a history of objects in the etcdbackup.operator.openshift.io.  You can order that history, read the jobs, and force-assign to the next master alphabetically.  If there is no history, randomly choose. You can force assign the job in the podtemplatespec.

For retention, i think a similar approach is possible by simply dividing the count (that's why count over size) and having the job itself delete the old backups (I'm betting this is your default flow).